### DISCUSSION TITLE
Setting custom pythonpath in a per project settings file

### DISCUSSION THREAD

**Discussion by rafaelleru** [python-lsp/python-lsp-server#177] [Configuration]
Hi all, sorry for asking here but I could not find the solution on jedi documentation neither on SO or pylsp docs.

Is there a way to custom set my pythonpath only for certain projects? I can do it via venv but I don't know how to do it with a config file in order to use it with the global LSP server in my machine.

Thanks.

**Comment by npradeep357** [comment]
@rafaelleru, What to do you mean by certain projects? Do you want to run this pylsp on a custom python (other than the globally installed one)?

**Comment by cbensimon** [comment]
@npradeep357 Is it actually be possible to run pylsp on a custom python ?
Thanks.

**Reply by npradeep357** [reply]
What do you mean by custom python? Is it venv? or custom python intrepretor (like PyPy, IronPython)?
if you mean venv, then yes, it is possible.

**Reply by npradeep357** [reply]
But if you mean a custom python interpreter, then I haven't tested it. You need to give it a try and see.

**Comment by ccordoba12** [comment]
Moving this to our new Discussions forum so you can keep talking about it there.

**Comment by skeledrew** [comment]
Hello. I was directed from [my issue](https://github.com/python-lsp/python-lsp-server/issues/178) to here for support. I have already resolved, but [sharing it](https://gitlab.com/-/snippets/2279333) here in case it helps someone.

My solution is to patch the Jedi interaction so it dynamically finds the venv associated with respective projects.

**Reply by HairlessVillager** [reply]
Your script is a real brainteaser! But still thanks!

**Reply by HairlessVillager** [reply]
I re-developed a Windows version script on yours. If anyone need, [click here](https://gist.github.com/HairlessVillager/5169fcfa8e38a9a3a8f0e197f8ddee3b).
